modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -650,6 +651,47 @@ public Page<ObjectEntry> getObjectEntryRelatedObjectEntries(
 				serviceBuilderObjectEntry.getPrimaryKey(), null));
 	}
 
+	@Override
+	public Page<ObjectEntry> getObjectEntryVersions(
+			DTOConverterContext dtoConverterContext, long objectEntryId,
+			Pagination pagination)
+		throws Exception {
+
+		return Page.of(
+			TransformUtil.transform(
+				_objectEntryVersionService.getObjectEntryVersions(
+					objectEntryId, _getStartPosition(pagination),
+					_getEndPosition(pagination)),
+				objectEntryVersion -> {
+					dtoConverterContext.setAttribute(
+						"objectEntryVersion", objectEntryVersion);
+
+					return _objectEntryDTOConverter.toDTO(
+						dtoConverterContext,
+						objectEntryLocalService.getObjectEntry(
+							objectEntryVersion.getObjectEntryId()));
```

Comment:
I think we can avoid performing this action `objectEntryLocalService.getObjectEntry()` on every iteration since we already have `objectEntryId` from the parameter.

Reply:
Good catch @marcelabc! done [here](https://github.com/liferay-bpm/liferay-portal/pull/4271/commits/0fb12b7aa9d822c1f355300e9d2129914c35d907)